collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -596,20 +602,24 @@ private void refreshView(boolean isGoingUp) {
                         break;
                     }
                     case FormEntryController.EVENT_REPEAT: {
-                        if (!formController.isGroupRelevant()) {
+                        FormEntryCaption fc = formController.getCaptionPrompt();
+                        boolean isFirst = fc.getMultiplicity() == 0;
+                        boolean forPicker = shouldShowRepeatGroupPicker();
+                        boolean isEmpty = !formController.isGroupRelevant();
+                        if (isEmpty &&
```

Comment:
I'm actually wondering if this could be `isEmpty && forPicker` instead? That also seems to work. I think we now only want to avoid the next piece of code in the case where we're avoiding rendering an "empty" group in the picker. I wouldn't be surprised if I've missed a case that you'd thought about, and if that is true, let's discuss how we get a test in to enforce this implementation.

Reply:
You're absolutely right, `isFirst` is unnecessary - it's tested further down so that non-empty repeats after the first are skipped anyway. I felt my updated logic was a bit fishy but couldn't spot exactly how.

To check this I've temporarily reinstated my original form and test (which initially caused difficulties - see below). I've simplified the code to match your suggestion, and both tests still pass. Tweak it further at L608 to restore the current logic, and both fail. 

The problem with the reinstatement seemed to be that the two test forms had different names but the same ID, breaking the whole test class in random ways. Adding __A_ to the ID of my form certainly fixed everything definitively.